src/remediations/fifi.js

```diff
@@ -43,6 +50,149 @@ exports.checkSmartManagement = async function (remediation, smart_management) {
+        rhcRuns[`count_${status}`] = _.size(_.filter(rhcRuns.data, run => run.status === status));
+    });
+
+    // Status of executor
+    rhcRuns.status = findRunStatus(rhcRuns);
+}
+
+exports.formatRunHosts = function (rhcRunHosts) {
+    return _.map(rhcRunHosts.data, host => ({
+        system_id: host.id,
+        system_name: 'localhost',
+        status: host.status,
+        updated_at: host.updated_at,
+        playbook_run_executor_id: exports.generateUuid()
+    }));
+};
+
+function formatRHCHostDetails (host, details) {
+    return {
+        system_id: host.id,
+        system_name: 'localhost',
+        status: host.status,
+        updated_at: host.updated_at,
+        console: details.data[0].stdout,
+        executor_id: exports.generateUuid()
+    };
+}
+
+function pushRHCSystem (host, systems) {
+    systems.push({
+        system_id: host.system_id,
+        system_name: 'localhost',
+        status: host.status,
+        updated_at: host.updated_at,
+        playbook_run_executor_id: exports.generateUuid()
+    });
+}
+
+function pushRHCExecutor (rhcRun, satRun) {
+    satRun.executors.push({
+        executor_id: exports.generateUuid(),
+        executor_name: 'Direct connected',
+        status: rhcRun.status,
+        system_count: rhcRun.system_count,
+        playbook_run_id: rhcRun.data[0].labels['playbook-run'],
+        playbook: rhcRun.data[0].url,
+        updated_at: rhcRun.data[0].updated_at,
+        count_failure: rhcRun.count_failure,
+        count_success: rhcRun.count_success,
+        count_running: rhcRun.count_running,
+        count_pending: 0, // RHC does not return the status pending
+        count_canceled: 0 // RHC does not currently return status canceled
+    });
+}
+
+exports.getRHCRuns = async function (playbook_run_id = null) {
+    const filter = createDispatcherRunsFilter(playbook_run_id);
+    const rhcRuns = await dispatcher.fetchPlaybookRuns(filter, RUNSFIELDS);
+
+    return rhcRuns;
+};
+
+exports.getRunHostDetails = async function (playbook_run_id, system_id) {
+    const runsFilter = createDispatcherRunHostsFilter(playbook_run_id);
+    const runHostsFilter = createDispatcherRunHostsFilter(playbook_run_id, system_id);
+    const [rhcRunHosts, rhcRunHostDetails] = await Promise.all([
+        dispatcher.fetchPlaybookRuns(runsFilter, RUNSFIELDS),
+        dispatcher.fetchPlaybookRunHosts(runHostsFilter, RUNHOSTFIELDS)
+    ]);
+
+    if (!rhcRunHosts || !rhcRunHostDetails) {
+        return null;
+    }
+
+    const host = _.find(rhcRunHosts.data, host => host.id === system_id);
+
+    return formatRHCHostDetails(host, rhcRunHostDetails);
+};
+
+exports.combineHosts = function (rhcRunHosts, systems) {
+    rhcRunHosts = exports.formatRunHosts(rhcRunHosts);
+
+    _.forEach(rhcRunHosts, host => {
+        pushRHCSystem(host, systems);
+    });
+};
+
+exports.combineRuns = async function (remediation) {
+    for (const satRun of remediation.playbook_runs) {
```

Comment:
@jharting Based on what needs to be returned (at least for the "get all playbook runs" method) I'm not sure if there's a way to identify which one is an RHC run before sending the request to the dispatcher since it's possible that each "run" has RHC systems included.